external-crates/move/crates/move-analyzer/src/completions/utils.rs

```diff
@@ -113,6 +303,53 @@ pub fn call_completion_item(
     }
 }
 
+// Constructs a cursor context and from existing symbols and
+// updates symbols to reflect this.
+pub fn compute_cursor(
+    symbols: &mut Symbols,
+    compiled_pkg_info: &mut CompiledPkgInfo,
+    cursor_path: &PathBuf,
+    cursor_pos: Position,
+) {
+    let cursor_info = Some((cursor_path, cursor_pos));
+    let mut symbols_computation_data = SymbolsComputationData::new();
+    let mut symbols_computation_data_deps = SymbolsComputationData::new();
+    // we only compute cursor context and tag it on the existing symbols to avoid spending time
+    // recomputing all symbols (saves quite a bit of time when running the test suite)
+    let mut cursor_context = compute_symbols_pre_process(
+        &mut symbols_computation_data,
+        &mut symbols_computation_data_deps,
+        compiled_pkg_info,
+        cursor_info,
+    );
+    cursor_context = compute_symbols_parsed_program(
+        &mut symbols_computation_data,
+        &mut symbols_computation_data_deps,
+        compiled_pkg_info,
+        cursor_context,
+    );
+    symbols.cursor_context = cursor_context;
+}
+
+pub fn addr_to_ide_string(addr: &Address) -> String {
```

Comment:
We have to make sure that for numerical addresses we only print either the name (if present) or address, rather than the combination of the two which is what implementation of `std::Display` for address does